### ISSUE TITLE
Cookies validation

### ISSUE THREAD

**Issue by li-ar** [yiisoft/yii2-httpclient#106]
Hi there!

I try to set cookies from previous response to new request. And i get this error
`Invalid Configuration – yii\base\InvalidConfigException
Cookie 'some-cookie-name' value 'some-cookie-valie-with-@-symbol' is invalid`
Cookie value is "somesecretname@gmail.com"
It's because of cookies value checking in composeCookieHeader method https://github.com/yiisoft/yii2-httpclient/blob/master/Request.php#L478
So, there are two checks - for cookie name and cookie value. And all of them use same method "validateCookieValue" which do not allow "@" symbol. And question is - why? ) According to RFC 6265  we have 

> The VALUE is opaque to the user agent and may be anything the origin server chooses to send, possibly in a server-selected printable ASCII encoding. 

Maybe there is should be two separated method for name/value check? Or I'm wrong and symbol '@' do not allowed at all in cookie values?

| Q                       | A
| ----------------------- | ---
| Yii version             | 2.0.12
| Yii HTTP Client version | 2.0.0
| PHP version             | 7.0.12
| Operating system        | FreeBSD 10.3-RELEASE

**Comment by klimov-paul** [comment]
```
cookie-octet      = %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
                       ; US-ASCII characters excluding CTLs,
                       ; whitespace DQUOTE, comma, semicolon,
                       ; and backslash
```

**Comment by klimov-paul** [comment]
Resolved by commit a0b5ff6e61fbb73cef7f7f7f59ace1c144db550a
